doc/pcre2api.3

```diff
@@ -1748,81 +1851,52 @@ though the reference can be by name or by number.
 .sp
   PCRE2_NO_AUTO_POSSESS
 .sp
-If this option is set, it disables "auto-possessification", which is an
-optimization that, for example, turns a+b into a++b in order to avoid
+If this (deprecated) option is set, it disables "auto-possessification", which
+is an optimization that, for example, turns a+b into a++b in order to avoid
 backtracks into a+ that can never be successful. However, if callouts are in
 use, auto-possessification means that some callouts are never taken. You can
 set this option if you want the matching functions to do a full unoptimized
 search and run all the callouts, but it is mainly provided for testing
 purposes.
+.P
+It is recommended to use \fBpcre2_set_optimize\fP with the \fIdirective\fP
+PCRE2_AUTO_POSSESS_OFF rather than the compile option PCRE2_NO_AUTO_POSSESS.
+Note that PCRE2_NO_AUTO_POSSESS takes precedence over the
+\fBpcre2_set_optimize\fP optimization directives PCRE2_AUTO_POSSESS and
+PCRE2_AUTO_POSSESS_OFF.
```

Comment:
rather something like:

```
If a compile context is available, it is recommended to use \fBpcre2_set_optimize()\fP with the PCRE2_AUTO_POSSESS_OFF \fIdirective\f
rather than this compile option.
Note that this option takes precedence over the \fBpcre2_set_optimize()\fP optimization directives PCRE2_AUTO_POSSESS and PCRE2_AUTO_POSSESS_OFF.
```

Then a very similar paragraph could be also used for the next option.

Goes without saying that using `pcre2_set_optimize()` might be more consistent as well with the other suggested API calls which were recently updated.

Reply:
"If a compile context is available" does make sense, updated.